realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -1183,17 +1183,17 @@ JNIEXPORT jlongArray JNICALL Java_io_realm_internal_TableQuery_nativeBatchUpdate
 
         // Step3: Run & export the queries against the latest shared group
         for (size_t i = 0; i < number_of_queries; ++i) {
-            JniLongArray query_param_array(env, (jlongArray) env->GetObjectArrayElement(query_param_matrix, i));
-            switch (query_param_array[0]) { // 0, index of the type of query, the next indicies are parameters
+            JniLongArray* query_param_array = new JniLongArray(env, (jlongArray) env->GetObjectArrayElement(query_param_matrix, i));
```

Comment:
Use `unique_ptr` to avoid deletion especially for the `return NULL` case. Like:
`std::unique_ptr<JniLongArray> query_param_array(new JniLongArray(env, (jlongArray) env->GetObjectArrayElement(query_param_matrix, i));`